src/lib/utils.js

```diff
@@ -74,7 +74,9 @@ let utils = {
       if (typeof key === 'string') {
         key = key.toLowerCase();
       }
-      if (!newCollection[key]) {
+      // fix #171 the given data type of docsearch hits might be conflict with the properties of the native Object,
+      // such as the constructor, so we need to do this check.
+      if (!Object.prototype.hasOwnProperty.call(newCollection, key)) {
```

Comment:
the test failed because you based on `master`, not `develop`. You can change it in the GitHub UI just under your title if you click on edit